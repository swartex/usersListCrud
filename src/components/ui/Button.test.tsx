import { render, screen, waitFor } from "@testing-library/react";
import { Button } from "./button";

describe("Button", () => {
  it("renders", () => {
    const { asFragment } = render(<Button>test</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders ghost button", () => {
    const { asFragment} = render(<Button variant="ghost">test1</Button>);
    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('render small button', () => {
    render(<Button size="sm" variant="destructive">Button text</Button>);
    const button = screen.findByRole('button');

    waitFor(() => {
      console.log(button)
    })
  })
});
