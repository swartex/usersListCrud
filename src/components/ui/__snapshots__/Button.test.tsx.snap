// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`Button > renders ghost button 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
  >
    test1
  </button>
</DocumentFragment>
`;
